Sources/Mustache/ContentType.swift

```diff
@@ -36,21 +38,32 @@ struct HTMLContentType: MustacheContentType {
 ///
 /// The string is read from the "CONTENT_TYPE" pragma `{{% CONTENT_TYPE: type}}`. Replace type with
 /// the content type required. The default available types are `TEXT` and `HTML`. You can register your own
-/// with `MustacheContentTypes.register`.
-public enum MustacheContentTypes {
-    static func get(_ name: String) -> MustacheContentType? {
-        self.types[name]
+/// with `MustacheContentTypes.shared.register`.
+public class MustacheContentTypes: @unchecked Sendable {
+
+    public static let shared = MustacheContentTypes()
+
+    private init() {}
+
+    private let lock = NSLock()
+
+    func get(_ name: String) -> MustacheContentType? {
+        lock.withLock {
+            self.types[name]
+        }
     }
 
     /// Register new content type
     /// - Parameters:
     ///   - contentType: Content type
     ///   - name: String to identify it
-    public static func register(_ contentType: MustacheContentType, named name: String) {
-        self.types[name] = contentType
+    public func register(_ contentType: MustacheContentType, named name: String) {
+        lock.withLock {
+            self.types[name] = contentType
+        }
     }
 
-    static var types: [String: MustacheContentType] = [
```

Comment:
> Static property 'types' is not concurrency-safe because it is nonisolated global shared mutable state

This error is appearred by compiler.
Normally we should use actor for sharing global states.

However there are very few cases where MustacheContentTypes are written at the same time, and we did not think it would be desirable to use an Async Context just for that. Therefore, we are using a Singleton with a lock on write.

Reply:
All you've done here is replace one static variable for another ie `shared`. Given access to this is protected via a lock you can add `nonisolated(unsafe)` to it. Remember though this is only available when `compiler >= 6.0`

Reply:
Makes sense. I reverted my changes 1f43e97a6bd4e63e4dc63f595d0560713b82d149 and use `nonisolated(unsafe)` above `compiler >= 6.0` 63084a2.

Reply:
56644a4 Support a lock in Swift 6 mode.

Reply:
You can still use the lock in earlier versions of Swift you just can't use `nonisolated(unsafe)`

Reply:
Okay, I applied lock to earlier versions. b254e2d